lib/agreements.js

```diff
@@ -1,42 +1,97 @@
 'use strict';
 
 var _ = require('lodash');
+var Joi = require('joi');
 
 exports.register = function (server, options, next) {
-  options = _.extend({ basePath: '/api/2.0' }, options);
+  options = _.extend({basePath: '/api/2.0'}, options);
   var handlers = require('./handlers.js')(server, 'cd-agreements');
 
   server.route([{
     method: 'POST',
     path: options.basePath + '/agreements',
     handler: handlers.actHandler('save'),
     config: {
-      description: 'Save the agreement',
-      tags: ['api', 'users']
+      description: 'Saves the specified agreement',
+      tags: ['api', 'users'],
+      plugins: {
+        'hapi-swagger': {
+          responseMessages: [
+            {code: 400, message: 'Bad Request'},
+            {code: 200, message: 'OK'}
+          ]
+        }
+      },
+      validate: {
+        payload: Joi.object({
+          agreement: {
```

Comment:
This is too restrctive, the payload used to save an agreement is 
{ agreementVersion: 2
  fullName: "derp"
  userId: "8735c503-bd3a-433f-9337-afd789cb7fec" }